### PR TITLE
Flow: fix VirtualizedList issues, remove few no longer valid suppressions

### PR DESCRIPTION
**Pull request by Simek**
## Summary

This PR fixes few suppressed Flow issues in `VirtualizedList` component.

I have also removed few no longer valid Flow suppressions in `ActivityIndicator` and `ReactNativeTestTools`.

## Changelog

[General] [Fixed] - Flow: fix few issues in `VirtualizedList`, 
[General] [Removed] - Flow: remove few no longer valid suppressions in `ReactNativeTestTools`

## Test Plan

Successful run of `yarn flow` check.

### CODE REVIEW

**Comment by Simek** [comment]
I had to revert the `ActivityIndicator` suppression because Flow check fails on the CI (but work locally). 

This might be something to check out and verify. Is it possible that CI uses different version of Flow?

File: Libraries/Utilities/ReactNativeTestTools.js [hunk 1]
```diff
@@ -57,13 +57,7 @@ function byClickable(): Predicate {
       // HACK: Find components that use `Pressability`.
       node.instance?.state?.pressability != null ||
       // TODO: Remove this after deleting `Touchable`.
-      /* $FlowFixMe(>=0.125.1 site=react_native_fb) This comment suppresses an
-       * error found when Flow v0.125.1 was deployed. To see the error, delete
-       * this comment and run Flow. */
       (node.instance &&
```
Comment (lunaleaps) [inline Libraries/Utilities/ReactNativeTestTools.js:60]: There are still flow errors when I imported this -- to fix, please change this to:
```
 (node.instance != null &&
        // $FlowFixMe[prop-missing] // Remove this after deleting `Touchable`
     typeof node.instance.touchableHandlePress === 'function'),
```

File: Libraries/Utilities/ReactNativeTestTools.js [hunk 2]
```diff
@@ -78,9 +72,6 @@ function byTestID(testID: string): Predicate {
 
 function byTextMatching(regex: RegExp): Predicate {
   return withMessage(
-    /* $FlowFixMe(>=0.125.1 site=react_native_fb) This comment suppresses an
-     * error found when Flow v0.125.1 was deployed. To see the error, delete
-     * this comment and run Flow. */
     node => node.props && regex.exec(node.props.children),
```
Comment (lunaleaps) [inline Libraries/Utilities/ReactNativeTestTools.js:75]: and this to 
```
node => node.props != null && regex.exec(node.props.children) !== null,
```

**Review by lunaleaps — CHANGES_REQUESTED** [review]

File: Libraries/Utilities/ReactNativeTestTools.js (hunk 2 above)
Comment (Simek) [reply Libraries/Utilities/ReactNativeTestTools.js:75]: @lunaleaps The `!=` in first check is a typo?

**Comment by Simek** [comment]
@lunaleaps Thank you for the review, I have applied the suggested changes.

**Pull request closed without merging**